src/plugins/data/server/search/search_service.ts

```diff
@@ -140,6 +152,58 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
       ) => {
         return this.search(context, searchRequest, options);
       },
+      searchSource: {
+        asScoped: async (request: KibanaRequest) => {
```

Comment:
Accepting a `KibanaRequest` was pretty much my only option for the API since it's what index patterns require on the server. The question (see below) is whether it's necessary to _also_ require `RequestHandlerContext` for the low-level search, when all of those items can already be derived from a `KibanaRequest`